pkg/openmetadata-connector-core/connector_helper.go

```diff
@@ -162,20 +228,28 @@ func NewOpenMetadataAPIService(conf map[interface{}]interface{}, logger *zerolog
 		NumRetries = DefaultNumRetries
 	}
 
-	nameToDatabaseStruct := make(map[string]database_types.DatabaseType)
-	nameToDatabaseStruct[MysqlLowercase] = database_types.NewMysql(logger)
-	nameToDatabaseStruct[S3] = database_types.NewS3(vaultConf, logger)
+	value, ok = conf["openmetadata_connector_port"]
+	if ok {
+		port = value.(int)
+	} else {
+		port = DefaultListeningPort
+	}
+
+	nameToDatabaseStruct := make(map[string]dbtypes.DatabaseType)
+	nameToDatabaseStruct[MysqlLowercase] = dbtypes.NewMysql(logger)
+	nameToDatabaseStruct[S3] = dbtypes.NewS3(vaultConf, logger)
 
 	s := &OpenMetadataAPIService{Endpoint: conf["openmetadata_endpoint"].(string),
 		SleepIntervalMS:      SleepIntervalMS,
 		NumRetries:           NumRetries,
 		NameToDatabaseStruct: nameToDatabaseStruct,
 		logger:               logger,
-		NumRenameRetries:     DefaultNumRenameRetries}
+		NumRenameRetries:     DefaultNumRenameRetries,
+		customization:        customization}
 
-	s.initialized = s.prepareOpenMetadataForFybrik()
+	s.initialized = s.PrepareOpenMetadataForFybrik()
 
-	return s
+	return s, port
```

Comment:
you don't have to return port (it is part of the service,. you can provide  access to it)